### PR TITLE
Clear pending_approval promptly on approval; fix codex stop-classifier

### PR DESCRIPTION
**Pull request by victorarias**
## Why

Two related "the attention light lies to you" bugs:

1. **Approval doesn't clear the light.** No Claude/Codex hook fires when the user *approves* a permission request. The onset is hook-driven (`PermissionRequest`/`permission_request` → `pending_approval`), but the only signal back to `working` is `PostToolUse`, which fires *after the approved tool finishes*. So a session sat in the flashing "needs you" state for the entire duration of the approved command — exactly when it didn't need you.
2. **Codex turns end purple.** The stop-time classifier (`codex exec`) refuses to run outside a trusted git dir, so a Codex turn ending in e.g. `/tmp` failed to classify and fell back to `unknown` (purple).

## What

**Approval → working, off the rendered PTY screen** (the interaction is only visible in the PTY):
- New `approvalResolver` (`internal/pty/approval_resolver.go`): arms on first sight of the approval prompt and emits `working` once the prompt has been gone for a 750ms debounce (so a repaint or a chained multi-step approval doesn't flap).
- It re-emits `pending_approval` on arm to sync the **pty-worker's own dedup view**. The worker only forwards a state when it differs from its last PTY-emitted state, and the onset hook sets pending *behind the worker's back* — without the re-emit, a chained second approval's `working` clear looks unchanged and gets dropped. (This was the day-long bug.)
- Detection uses the vt10x `virtualScreen.renderedText()` (what the user actually sees), not the raw byte tail — the tail keeps scrollback and re-matches the stale prompt.
- Wired via a new `HasApprovalResolver` capability (claude + codex), env-overridable like the other caps. Codex `ShouldApplyPTYState` now allows **only** the single `pending_approval → working` transition; all other live state stays hook-owned.

**Codex stop-classifier fixes** (`internal/classifier`):
- `--skip-git-repo-check` — classify regardless of the session's cwd (the actual unknown-state fix).
- `--ignore-user-config` — don't inherit the user's `config.toml`: no MCP servers (latency, auth noise, tool-schema tokens) or other agent settings. Auth is read separately and still works. (`-c mcp_servers={}` merges and does nothing; this is the clean global disable.)
- `-c features.shell_tool=false` — the classifier only emits a verdict, never runs commands.
- Single model `gpt-5.4-mini` (low effort), fallback list removed. Override via `ATTN_CODEX_CLASSIFIER_MODEL`. (`gpt-5.3-codex` isn't available on a ChatGPT-account login.)
- Net: ~26.4k → ~22.5k input tokens per classification, no honeycomb MCP startup.

## Testing

- `go build ./...` clean; `go test ./internal/classifier ./internal/pty ./internal/agent` green.
- `approval_resolver_test.go`: arm emits pending, clear-after-debounce emits working, repaint resets the debounce, no transition without a prompt.
- `daemon_policy_test.go`: codex applies `working` to clear pending, ignores every other transition out of pending.
- `classifier_test.go`: single invocation with the expected model and that the args carry `--skip-git-repo-check`, `--ignore-user-config`, `features.shell_tool=false`.
- Resolver verified live earlier for both Claude and Codex (fast PTY-driven pending→working, not a slow post-tool one). Codex classifier verified at the CLI level (exact code-produced invocation returns the right verdict with 0 MCP lines).

## Notes for review
- macOS-only, per repo policy.
- CHANGELOG updated under `## [2026-06-03]`.
- The classifier runs in the **daemon** process (needs a daemon restart to exercise end-to-end); the resolver runs in the per-session **worker** (new sessions pick it up automatically).

### CODE REVIEW

**Review by figgyster — DISMISSED** [review]
This needs another pass before merge.

Two blockers:

- The approval resolver is still output-driven. `resolveApproval` only runs from `readLoop` when more PTY bytes arrive; after the prompt disappears it records `clearedSince`, but the 750ms debounce can only complete on a later read. If the approved command is quiet for a long time — exactly the painful case this PR is trying to fix — the session can stay flashing `pending_approval` until the command prints again or exits. The timer needs to be driven independently once the prompt has disappeared, or the first prompt-gone sample needs to schedule the clear.
- CI is red on formatting: `internal/pty/approval_resolver_test.go` needs `gofmt` (`TestApprovalResolver_PromptReappearsResetsDebounce`, the aligned inline comments). Tiny rake, but it blocks the daemon job.

Local check I ran: `git diff --check` and `go test ./internal/classifier ./internal/pty ./internal/agent` passed in the PR checkout. The GitHub daemon job is failing earlier on `gofmt`.

**Review by figgyster — DISMISSED** [review]
This still needs one fix before merge.

The formatting blocker is gone and the focused local checks pass now (`git diff --check`; `go test ./internal/classifier ./internal/pty ./internal/agent`). But the approval clear is still tied to future PTY samples: once the prompt disappears, `clearedSince` is recorded, and the debounce only completes if another screen sample arrives later. A quiet approved command can still leave the session flashing `pending_approval` until it prints or exits — the exact contract this PR is trying to fix.

Required before merge:
- Drive the clear with an independent timer/scheduled recheck once the prompt is first absent, or otherwise emit `working` without requiring later PTY output.

**Review by figgyster — DISMISSED** [review]
This still needs one fix before merge. The follow-up added the rendered-screen resolver, but it is still only driven from `readLoop` when new PTY bytes arrive.

That means the exact quiet-command case can still stick in `pending_approval`: the prompt disappears, `clearedSince` is set on that output frame, and then nothing calls `observe()` again after `approvalClearDebounce` unless the approved command prints another byte. The new tests walk time forward by calling `observe()` manually, but the production path has no timer/ticker/scheduled callback to make that second call.

Required before merge:
- add a timer/ticker/scheduled recheck after the prompt first disappears, or otherwise make the debounce completion independent of later PTY output
- add a regression test that models “prompt gone, then no further PTY output” and proves `working` is emitted promptly

I did run the focused checks here: `git diff --check origin/main...HEAD` and `go test ./internal/pty ./internal/agent ./internal/classifier` both pass. The blocker is behavioral, not build noise.

**Review by figgyster — DISMISSED** [review]
This still needs one fix before merge.

The follow-up cleared the formatting failure, and the focused local checks pass (`git diff --check origin/main...HEAD`; `go test ./internal/pty ./internal/agent ./internal/classifier`). The remaining blocker is the same approval-clear contract: production still only calls the resolver from PTY output handling, so after the prompt disappears the debounce cannot complete unless another byte arrives later. A silent approved command can still sit in `pending_approval` until it prints or exits.

Required before merge:
- make the clear independent of future PTY output, e.g. a timer/ticker/scheduled recheck after the prompt first disappears
- add a regression test for “prompt disappears, then no further PTY output” proving `working` is emitted promptly

**Review by figgyster — DISMISSED** [review]
This still needs one fix before merge. The Codex classifier side looks fine, and the focused Go tests pass locally, but the approval clear is still output-driven.

`Session.resolveApproval()` only runs from `readLoop` when more PTY bytes arrive. So if the approval prompt disappears and the approved command then goes quiet for a long stretch, the resolver records/starts the debounce only on that last repaint and never gets called again after 750ms. The light stays in `pending_approval` until the command emits another byte — which is exactly the bug this PR is trying to kill.

Required before merge:
- Drive the prompt-gone debounce from a timer/ticker/scheduled callback, or otherwise re-sample the rendered screen after `approvalClearDebounce` without depending on further PTY output.
- Add a regression test for the silent approved-command case: prompt visible → prompt gone → no more PTY bytes → `working` emits promptly after the debounce.

Local checks run: `git diff --check origin/main...HEAD` and `go test ./internal/pty ./internal/classifier ./internal/agent`.

**Review by figgyster — DISMISSED** [review]
This still needs one fix before merge. The Codex classifier side looks fine, and the focused Go tests pass locally, but the approval clear is still output-driven.

`Session.resolveApproval()` only runs from `readLoop` when more PTY bytes arrive. So if the approval prompt disappears and the approved command then goes quiet for a long stretch, the resolver records/starts the debounce only on that last repaint and never gets called again after 750ms. The light stays in `pending_approval` until the command emits another byte — which is exactly the bug this PR is trying to kill.

Required before merge:
- Drive the prompt-gone debounce from a timer/ticker/scheduled callback, or otherwise re-sample the rendered screen after `approvalClearDebounce` without depending on further PTY output.
- Add a regression test for the silent approved-command case: prompt visible → prompt gone → no more PTY bytes → `working` emits promptly after the debounce.

Local checks run: `git diff --check origin/main...HEAD` and `go test ./internal/pty ./internal/classifier ./internal/agent`.

**Review by figgyster — DISMISSED** [review]
This still needs the same behavioral fix before merge.

The new head adds the resolver plumbing and the focused checks pass, but production is still output-driven: `Session.resolveApproval()` is only called from `readLoop` when PTY bytes arrive. After the prompt disappears, the resolver records `clearedSince`; if the approved command then goes quiet, nothing calls `observe()` again after the 750ms debounce. The light can still stay in `pending_approval` until the command prints or exits — the bug wearing a new hat.

Required before merge:
- drive the prompt-gone debounce with a timer/ticker/scheduled recheck, or otherwise make `working` emission independent of later PTY output
- add a regression test at the production boundary for “prompt visible → prompt gone → no more PTY bytes → `working` emitted promptly”

Local checks run: `git diff --check origin/main...HEAD` and `go test ./internal/pty ./internal/classifier ./internal/agent`.

**Review by figgyster — APPROVED** [review]
This looks safe to merge now. The approval clear is no longer hostage to later PTY output — the timer-backed resolver covers the quiet approved-command case — and the Codex classifier changes avoid the trusted-git-dir/config bleed without dragging user MCP plumbing into a stop classifier. Good fix for two “the light lies” bugs, which is exactly the kind of gremlin this app is meant to kill.

Validated locally:
- `git diff --check`
- `go test ./internal/pty ./internal/classifier ./internal/agent ./internal/daemon`

GitHub checks are green from here. No blockers from me.

File: internal/pty/approval_resolver.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		return "", false
+	}
+
+	if !r.armed {
+		return "", false
+	}
+
+	if r.clearedSince.IsZero() {
+		r.clearedSince = now
+		return "", false
```
[the remark this answers is not in the document]
Comment (victorarias) [reply internal/pty/approval_resolver.go]: Addressed in 7d2fcb43 — this is exactly what I implemented. The prompt-gone sample now returns `approvalClearStarted`, and the session schedules a one-shot `time.AfterFunc(debounce + margin)` recheck, so a silent/long approved command clears to `working` (~0.85s) without needing any further PTY output. Covered by `TestSession_ApprovalClearsWithoutFurtherOutput` (prompt → gone → no output → working, green under `-race`).

— Claude, on behalf of Victor
